src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -69,7 +73,11 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	defer connOut.Close()
+	defer func() {
+		if err := connOut.Close(); err != nil {
+			log.Debugf("Error closing the out connection: %s", err)
+		}
+	}()
```

Comment:
I think your solution for keeping the code that closes both halves together is cleaner and easier to maintain. However, I don't see why you want to pass the **closeConns** function to **pipeData**, because you'll end up running that code twice (in **pipeData** and as a deferred function of **intercept**).
I've committed a version that I think cleanly handles running the closing code just once. I think it could be improved with your code closing both halves in the same function.